tests/SushiTest.php

```diff
@@ -2,104 +2,157 @@
-        );
+        $this->assertEquals(101, Maki::count());
     }
 
     /** @test */
-    function uses_same_cache_between_requests()
+    public function it_uses_correct_attributes()
     {
-        $this->markTestSkipped('I can\' find a good way to test this right now');
+        $model = Maki::whereKey(1)->first();
+
+        $this->assertSame(1, $model->id);
+        $this->assertSame('foobar', $model->name);
+        $this->assertSame(10, $model->age);
+        $this->assertSame(9.99, $model->price);
+        $this->assertSame(['lorem', 'ipsum'], $model->properties);
+        $this->assertInstanceOf(DateTime::class, $model->published_at);
+        $this->assertSame('2020-01-25 19:45:30', $model->published_at->format('Y-m-d H:i:s'));
+        $this->assertInstanceOf(DateTime::class, $model->created_at);
+        $this->assertInstanceOf(DateTime::class, $model->updated_at);
     }
 
-    /** @test */
-    function use_same_cache_between_requests()
+    protected function assertEqualsColumnDefinition(array $expected, $actual)
     {
-        $this->markTestSkipped('I can\' find a good way to test this right now');
+        $this->assertInstanceOf(ColumnDefinition::class, $actual);
+        foreach($expected as $key => $value) {
+            $this->assertSame($value, $actual[$key]);
+        }
     }
 }
 
-class Foo extends Model
+class Maki extends Model
 {
-    use \Sushi\Sushi;
+    use Sushi;
 
-    protected $rows = [
-        ['foo' => 'bar', 'bob' => 'lob'],
-        ['foo' => 'baz', 'bob' => 'law'],
+    protected $guarded = [];
+
+    public $casts = [
+        'age' => 'int',
+        'price' => 'float',
+        'properties' => 'array',
+        'published_at' => 'datetime',
     ];
 
-    public static function resetStatics()
-    {
-        static::setSushiConnection(null);
-        static::clearBootedModels();
-    }
+    protected const ROWS = [
+        ['id' => 1, 'name' => 'foobar', 'age' => 10, 'price' => 9.99, 'properties' => ['lorem', 'ipsum'], 'published_at' => '2020-01-25 19:45:30'],
+        ['id' => 2, 'name' => 'minion', 'age' => 20, 'price' => 19.99, 'properties' => ['minion', 'banana'], 'published_at' => '2020-01-28 08:15:20'],
+    ];
 
-    public static function setSushiConnection($connection)
+    public static function getRows(): LazyCollection
     {
-        static::$sushiConnection = $connection;
-    }
-}
+        $faker = Factory::create();
 
-class Bar extends Model
-{
-    use \Sushi\Sushi;
+        return LazyCollection::range(0, 100)->map(function (int $i) use ($faker): array {
+            $row = self::ROWS[$i] ?? [
+                'name' => $faker->word,
+                'age' => $faker->numberBetween(1, 99),
+                'price' => $faker->randomFloat(2, 1, 1000),
+                'properties' => $faker->words(),
+                'published_at' => $faker->dateTimeThisDecade,
+            ];
+
+            $row['published_at'] = Carbon::make($row['published_at']);
```

Comment:
Datetime fields already accept any \DatetimeInteface. Converting a \Datetime to a \Carbon does not make sense.

Reply:
This line is only for the correct database column type. 
```php
if (is_object($value) && $value instanceof DateTime) {
    $type = 'timestamp';
}
```

Because the first row is the hardcoded one with a simple string.

Reply:
`$row['published_at'] = Carbon::make($row['published_at']);` is useless.

Reply:
It's not, the unittests will fail without because the `published_at` is only a string for the first two rows and therefor it's not mapped to `timestamp` column type.